Back End Challenge/src/main/java/com/test/headstorm/challenge/controller/RandomNumbersController.java

```diff
@@ -0,0 +1,41 @@
+package com.test.headstorm.challenge.controller;
+
+import com.test.headstorm.challenge.service.RandomNumbersService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.*;
+
+import javax.annotation.Resource;
+import java.util.List;
+
+/**
+ * <p>  </p>
+ *
+ * @author
+ * @version $Id: RandomNumbersController.java, v 0.1 2019/8/15  Exp $
+ */
+@RestController
+@RequestMapping(value="/data")
+public class RandomNumbersController {
+    private Logger logger = LoggerFactory.getLogger(RandomNumbersController.class);
+
+    @Resource
+    RandomNumbersService randomNumbersService;
+
+    @PostMapping(value="/numbers", consumes = "application/json")
+    public String numbersCheck(@RequestBody List<Integer> reqList) {
+        return randomNumbersService.numbersCheck(reqList);
+    }
+
+    @GetMapping(value="/numbers", consumes = "application/json")
+    public List<Integer> numbersSort(@RequestBody List<Integer> reqList) {
+        return randomNumbersService.numbersSort(reqList);

```

Comment:
If we discover that this data service is very "read-heavy", or in other words, that our users POST new lists **very** rarely, but GET the number list **very** frequently, how would you modify or optimize this solution to perform less work and return faster responses to your users in this case?